# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.h
# Commit: 53373d35978a
# Full Hash: 53373d35978a5445344404737fc70f8929830102
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-03-12 03:55:25
# Regressor Bug: 1948339
# File Overlap Count: 2
# Description:
#   Bug 1948339 - Part 2: Add NotificationObserver r=asuth
#   
#   This also makes page-bound notifications to not be stored via NotificationStorage.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240642
# ==============================================================================

diff -r 96d954eae2a0 -r 53373d35978a dom/notification/NotificationParent.h
--- a/dom/notification/NotificationParent.h	Tue Mar 11 22:23:52 2025 +0000
+++ b/dom/notification/NotificationParent.h	Tue Mar 11 22:24:26 2025 +0000
@@ -15,14 +15,22 @@
 
 enum class CloseMode;
 
+enum class AlertTopic : uint8_t {
+  Disable,
+  Settings,
+  Click,
+  Show,
+  Finished,
+};
+
 class NotificationParent final : public PNotificationParent,
-                                 public nsIObserver {
+                                 public nsISupports,
+                                 public SupportsWeakPtr {
   using IPCResult = mozilla::ipc::IPCResult;
 
  public:
   // Threadsafe to pass the class from PBackground to main thread
   NS_DECL_THREADSAFE_ISUPPORTS
-  NS_DECL_NSIOBSERVER
 
   NotificationParent(NotNull<nsIPrincipal*> aPrincipal,
                      NotNull<nsIPrincipal*> aEffectiveStoragePrincipal,
@@ -35,6 +43,7 @@
         mScope(aScope),
         mOptions(aNotification.options()) {};
 
+  nsresult HandleAlertTopic(AlertTopic aTopic);
   IPCResult RecvShow(ShowResolver&& aResolver);
   IPCResult RecvClose();
 
@@ -42,16 +51,13 @@
       Endpoint<PNotificationParent>&& aParentEndpoint,
       PBackgroundParent::CreateNotificationParentResolver&& aResolver);
 
-  void ActorDestroy(ActorDestroyReason aWhy) override;
-
  private:
   ~NotificationParent() = default;
 
   nsresult Show();
   nsresult FireClickEvent();
-  nsresult FireCloseEvent();
 
-  void Unregister(CloseMode aCloseMode);
+  void Unregister();
 
   Maybe<NotificationParent::ShowResolver> mResolver;
 